src/client/content_mapblock.cpp

```diff
@@ -1819,3 +1822,300 @@ void MapblockMeshGenerator::generate()
+					while (u + u1 < BITSET_MAX_NOPAD && // while still in current chunk
+						(m_slices[slice_offset + u + u1] & mask) == mask) {
+						// and next column shares faces
+						m_slices[slice_offset + u + u1] ^= mask;
+						u1++;
+					}
+					const core::vector2d<f32> uvs[4] = {
+						core::vector2d<f32>{0, static_cast<f32>(v1*width)},
+						core::vector2d<f32>{0, 0},
+						core::vector2d<f32>{static_cast<f32>(u1*width), 0},
+						core::vector2d<f32>{static_cast<f32>(u1*width), static_cast<f32>(v1*width)}
+					};
+					u1 = (u + u1) * scaled_BS - BS / 2;
+					const s32 u0 = u * scaled_BS - BS / 2;
+					v1 = (v0 + v1) * scaled_BS - BS / 2;
+					v0 = v0 * scaled_BS - BS / 2;
+					const s32 w = ((slice_i + 1) * width - 1
+						+ (direction % 2 == 0 ? -width + 1 : 1)) * BS
+						- BS / 2;
+					switch (direction) {
+					case 0:
+						vertices[0] = core::vector3df(w, u0, v1);
+						vertices[1] = core::vector3df(w, u1, v1);
+						vertices[2] = core::vector3df(w, u1, v0);
+						vertices[3] = core::vector3df(w, u0, v0);
+						break;
+					case 1:
+						vertices[0] = core::vector3df(w, u0, v0);
+						vertices[1] = core::vector3df(w, u0, v1);
+						vertices[2] = core::vector3df(w, u1, v1);
+						vertices[3] = core::vector3df(w, u1, v0);
+						break;
+					case 2:
+					case 3:
+						vertices[0] = core::vector3df(u0, w, v0);
+						vertices[1] = core::vector3df(u1, w, v0);
+						vertices[2] = core::vector3df(u1, w, v1);
+						vertices[3] = core::vector3df(u0, w, v1);
+						break;
+					case 4:
+						vertices[0] = core::vector3df(u0, v0, w);
+						vertices[1] = core::vector3df(u0, v1, w);
+						vertices[2] = core::vector3df(u1, v1, w);
+						vertices[3] = core::vector3df(u1, v0, w);
+						break;
+					default:
+						vertices[0] = core::vector3df(u1, v0, w);
+						vertices[1] = core::vector3df(u0, v0, w);
+						vertices[2] = core::vector3df(u0, v1, w);
+						vertices[3] = core::vector3df(u1, v1, w);
+						break;
+					}
+					for (core::vector3df& v : vertices)
+						v += seg_offset;
+
+					switch (direction) {
+					case 0:
+					case 2:
+					case 4:
+						irr_vertices[0] = video::S3DVertex(vertices[0], s_normals[direction], color, uvs[0]);
+						irr_vertices[1] = video::S3DVertex(vertices[1], s_normals[direction], color, uvs[1]);
+						irr_vertices[2] = video::S3DVertex(vertices[2], s_normals[direction], color, uvs[2]);
+						irr_vertices[3] = video::S3DVertex(vertices[3], s_normals[direction], color, uvs[3]);
+						break;
+					default:
+						irr_vertices[0] = video::S3DVertex(vertices[0], s_normals[direction], color, uvs[0]);
+						irr_vertices[1] = video::S3DVertex(vertices[3], s_normals[direction], color, uvs[1]);
+						irr_vertices[2] = video::S3DVertex(vertices[2], s_normals[direction], color, uvs[2]);
+						irr_vertices[3] = video::S3DVertex(vertices[1], s_normals[direction], color, uvs[3]);
+					}
+					m_collector->append(m_is_mono_mat ? s_static_tile : tile, irr_vertices, 4, quad_indices, 6);
```

Comment:
wait what exactly does the LOD mesher do (compared to the normal one) if `!m_is_mono_mat`?